### DISCUSSION TITLE
Aren't handles just refs?

### DISCUSSION THREAD

**Discussion by tom-sherman** [unboundedsystems/adapt#94] [General]
Why this?

```js
import { handle } from "@adpt/core"

const pg = handle()

return <Postgres handle={pg} />
```

And not this?

```js
import { useRef } from "react"

const pg = useRef()

return <Postgres ref={pg} />
```

**Comment by mterrel** [comment]
Hey @tom-sherman! Thanks for the question.

Handles are definitely modeled after how React uses `ref`, so there are intentionally a lot of similarities. But handles aren't identical in all ways and have considerable additional behavior "behind the scenes". In particular, `useRef` wouldn't create what we need to make handles work. But even putting that aside, there are enough differences that we didn't think it made sense to shoehorn a "kinda similar" thing into `ref` and then have users be surprised about the differences.

**Reply by tom-sherman** [reply]
Interesting, thanks! I think this came from a misunderstanding of mine where I assumed that adapt was using React and a custom reconciler, but it's not right? More of a React-inspired programming model and not using React itself (like Ink, React Native, react-three)?

**Comment by mterrel** [comment]
Yes, that's exactly right. We use JSX/TSX and the basic conceptual model that React uses, but don't actually use any React code.

Side note:  when we first started this project, React was a little less easy to use for a use case like Adapt. I believe there have been some changes to React that might have us make that choice differently if we were to start from scratch today.
